src/fides/core/config/docs.py

```diff
@@ -0,0 +1,181 @@
+        )
+        return doc_string
+    except KeyError:
+        print(field_info)
+        raise SystemExit(f"!Failed to parse field: {field_name}!")
+
+
+def build_title_header(title: str) -> str:
+    """Build a pretty, TOML-valid title header."""
+    title_piece = f"#-- {title.upper()} --#\n"
+    top_piece = f"#{'-' * (len(title_piece) - 3)}#\n"
+    bottom_piece = f"#{'-' * 68}#\n"
+    header = top_piece + title_piece + bottom_piece
+    return header
+
+
+def convert_settings_to_toml_docs(settings_name: str, settings: BaseSettings) -> str:
+    """
+    Takes a Pydantic settings object and returns a
+    formatted string with included metadata.
+
+    The string is expected to be valid TOML.
+    """
+    settings_schema = settings.schema()
+    included_keys = set(settings.dict().keys())
+    title_header = build_title_header(settings_name)
+
+    # Build the Section docstring
+    settings_description = settings_schema["description"]
+    settings_docstring = f"[{settings_name}] # {settings_description}\n"
+
+    # Build the field docstrings
+    fields = remove_excluded_fields(settings_schema["properties"], included_keys)
+    field_docstrings = [
+        build_field_documentation(field_name, field_info)
+        for field_name, field_info in fields.items()
+    ]
+    full_docstring = (
+        title_header + settings_docstring + "\n" + "\n".join(field_docstrings)
+    )
+    return full_docstring
+
+
+def remove_excluded_fields(
+    fields: Dict[str, Dict], included_fields: Set[str]
+) -> Dict[str, Dict]:
+    """Remove fields that are marked as 'excluded=True' in their field."""
+    without_excluded_fields = {
+        key: value for key, value in fields.items() if key in included_fields
+    }
+    return without_excluded_fields
+
+
+def generate_config_docs(
+    config: FidesConfig, outfile_path: str = ".fides/fides.toml"
+) -> None:
+    """
+    Autogenerate the schema for the configuration file
+    and format it nicely as valid TOML.
+
+    _Anything at the top-level of the config is ignored!_
+    """
+
+    settings: Dict[str, BaseSettings] = get_nested_settings(config)
+    ordered_settings: Dict[str, BaseSettings] = {
+        name: settings[name] for name in sorted(set(settings.keys()))
+    }
+    nested_settings_docs: List[str] = [
+        convert_settings_to_toml_docs(settings_name, settings_schema)
+        for settings_name, settings_schema in ordered_settings.items()
+    ]
+    docs: str = "\n".join(nested_settings_docs)
+
+    # Verify it is valid TOML before writing it out
+    toml.loads(docs)
+
+    # assert (
+    #     "# TODO" not in docs
+    # ), "All fields require documentation, no description TODOs allowed!"
+
+    with open(outfile_path, "w", encoding="utf-8") as output_file:
+        output_file.write(docs)
+        print(f"Exported configuration file to: {outfile_path}")
+
+    # Verify it is a valid Fides config file
+    get_config(outfile_path)
```

Comment:
built-in test. If the file is confirmed good `toml`, then the config will throw an uncaught exception when it gets loaded and should warn us